### DISCUSSION TITLE
AOV layer names in multipass EXRs?

### DISCUSSION THREAD

**Discussion by nilsio** [p2or/blender-loom#99] [Ideas]
Hey just looking into how to marry Blender nicely with traditional Nuke based VFX pipeline. Loom looks very useful for this.

Is there any plan option possibility of also naming the layers within a multichannel EXR? So Nuke would see "depth" rather than "VIewLayer_Depth", and "rgba" rather than "ViewLayer_Combined" etc et. ?

Regards

**Comment by p2or** [comment]
Hi @nilsio,

thanks for your request. I feel for you, I'm also using a pretty ancient script in order to add a dedicated output node to compositor each time. I'm all in for a well thought, easy to use, unified way shipped with _Loom_ - the question is how to implement that properly in terms of UI and usability though...?  If you have a good idea or mockup to solve this, I would appreciate your feedback.

Technically we could either automate Blender's compositor (as my ancient script already does) or just rename the channels of the EXR after the rendering is done (need to investigate, involves a lot of guesswork probably). Besides the known channels, which passes would be nice to rename as well from your point of view? 

Cheers,
Christian

**Reply by nilsio** [reply]
Hey Christian,

Thanks for the speedy reply. We're just in the beginning stages of integrating Blender into our pipeline, EEVEE is one thing that's drawing us, plus younger talent being more Blender centric.

I was impressed how simple it was to align the colour management with Nuke and the passes it give mostly working out of the box too. Just some more control over how it outputs things seems needed, which is why your add-on appealed. I guess some of this could be handled by Deadline too? I'm just a compositor though so I guess our engineers will sort it out...

I did read that you can use the built in compositor to shuffle passes around, but just seemed a bit hacky. I think for the time being we'll likely create a Nuke Gizmo to shuffle channels around and rename them.

FWIW the passes I'd expect to receive would be:
rgba
diffuse
emission
specular
sss
reflection
shadows
normal
position
depth
motion
CryptoMaterial00

Then all the various lightgroups. And I suppose different facilities have different naming.

I'll get back to you once we're further along the road if I have anything to pass on.

Cheers,
Nils
